reconcile/quay_mirror.py

```diff
@@ -0,0 +1,107 @@
+from reconcile import queries
+
+from utils import gql
+from utils import secret_reader
+from utils import skopeo
+
+
+class QuayMirror:
+
+    QUAY_ORG_CATALOG_QUERY = """
+    {
+      quay_orgs: quay_orgs_v1 {
+        name
+        pushUser {
+          path
+          field
+        }
+        pushToken {
+          path
+          field
+        }
+      }
+    }
+    """
+
+    QUAY_REPOS_QUERY = """
+    {
+      apps: apps_v1 {
+        quayRepos {
+          org {
+            name
+          }
+          items {
+            name
+            mirror
+          }
+        }
+      }
+    }
+    """
+
+    def __init__(self, dry_run=False):
+        self.gqlapi = gql.get_api()
+        self.desired_state = self._get_desired_state()
+        self.push_creds = self._get_push_creds()
+        self.skopeo_cli = skopeo.Skopeo(dry_run)
+        self.settings = queries.get_app_interface_settings()
+
+    def run(self):
+        for org, data in self.desired_state.items():
+            creds = self.push_creds.get(org)
+            if creds is None:
+                raise RuntimeError(f'No push credentials found for Quay'
+                                   f'org {org}')
+
+            self.skopeo_cli.copy_all(src_image=data[0],
+                                     dst_image=data[1],
+                                     dest_creds=f'{creds[0]}:{creds[1]}')
+
+    def _get_desired_state(self):
+        result = self.gqlapi.query(self.QUAY_REPOS_QUERY)
+
+        state = {}
+
+        for app in result['apps']:
+            quay_repos = app.get('quayRepos')
+
+            if quay_repos is None:
+                continue
+
+            for quay_repo in quay_repos:
```

Comment:
i usually use `for quay_repo in quay_repos or []` to avoid checking for None explicitly

Reply:
Why to avoid checking for `None`? I mean, your suggestion is way more expensive (like ~50% more expesive) and, for my eyes, less readable. Mind if I keep it as is?

Reply:
don't mind at all! just consulting on things we've done in the past.